lib/routes/account.js

```diff
@@ -1004,26 +1078,35 @@ module.exports = function (
         log.begin('Account.RecoveryEmailResend', request)
         var sessionToken = request.auth.credentials
         var service = request.payload.service || request.query.service
-        if (sessionToken.emailVerified) {
+
+        // Choose which type of email to send
+        var code, func
+        if (sessionToken.emailVerified && sessionToken.tokenVerified) {
           return reply({})
+        } else if (!sessionToken.emailVerified || sessionToken.tokenVerified) {
+          code = sessionToken.emailCode
+          func = mailer.sendVerifyCode
+        } else {
+          code = sessionToken.tokenVerificationId
+          func = mailer.sendVerifyLoginEmail
```

Comment:
The branching logic here is not obvious to me.  What is `!sessionToken.emailVerified || sessionToken.tokenVerified` guarding against?  My naive impression is that sending the emailCode is never the right thing to do, but I could easily be missing something.